oauthproxy.go

```diff
@@ -343,6 +360,7 @@ func (p *OAuthProxy) buildProxySubrouter(s *mux.Router) {
 
 	// The userinfo and logout endpoints needs to load sessions before handling the request
 	s.Path(userInfoPath).Handler(p.sessionChain.ThenFunc(p.UserInfo))
+	s.Path(refreshPath).Handler(p.refreshChain.ThenFunc(p.Proxy))
```

Comment:
Hmm, why when you refresh will it then proxy you to the backend, is that exepected? Or should we just be returning some sort of static success response? I would expect that the refresh endpoint is not to be called by an average person in their browser but instead programatically?

Reply:
It's been a while since I wrote this up, but I think that I did this because:
1. The backend may not support and/or may have refresh turned off
2. If the user's token has expired past the refresh window

In both those cases I believe the proxy request would then see an invalid session and take the user back into a login cycle.

Reply:
So I'm thinking about how the `Auth` endpoint works, if you're authed, it returns 202, if you're not, it returns 403.

Should the refresh endpoint not behave the same?

For example, lets say I hit the refresh endpoint, and it correctly refreshes me, `p.Proxy` will then proxy you to the configured backend (IIRC), but at what path on the backend? It seems hard to define a correct behaviour there.

I had imagined this endpoint would be called in the same way the Auth endpoint is, and so the responses would be similar, and would have nothing to do with the backend server that we are proxying to?